Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -5627,6 +5628,77 @@ void WebGLRenderingContextBase::forceContextLost()
     forceLostContext(WebGLRenderingContextBase::RealLostContext);
 }
 
+static ASCIILiteral debugMessageTypeToString(GCGLenum type)
+{
+    switch (type) {
+    case GraphicsContextGL::DEBUG_TYPE_ERROR:
+        return "error"_s;
+    case GraphicsContextGL::DEBUG_TYPE_DEPRECATED_BEHAVIOR:
+        return "deprecated behavior"_s;
+    case GraphicsContextGL::DEBUG_TYPE_UNDEFINED_BEHAVIOR:
+        return "undefined behavior"_s;
+    case GraphicsContextGL::DEBUG_TYPE_PORTABILITY:
+        return "portability"_s;
+    case GraphicsContextGL::DEBUG_TYPE_PERFORMANCE:
+        return "performance"_s;
+    case GraphicsContextGL::DEBUG_TYPE_MARKER:
+        return "marker"_s;
+    case GraphicsContextGL::DEBUG_TYPE_OTHER:
+        return "other"_s;
+    default:
+        ASSERT_NOT_REACHED();
+        return "unknown"_s;
+    }
+}
+
+static ASCIILiteral debugIdToString(GCGLenum id)
```

Comment:
this already exits as errorCodeToString

Reply:
E.g. the spec is that when source is api, type is error, id is of type gl error enum

Reply:
> this already exits as errorCodeToString

But `GCGLErrorCode` is an bitfield not a gl error enum, so we still need `glErrorToErrorCode`.

I chose `debugIdToString` to keep with the pattern of the similar functions.